regex_helper.c

```diff
@@ -0,0 +1,86 @@
+#include <ctype.h>
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+#include <syslog.h>
+#include <stdlib.h>
+
+#include "user_secret.h"
+#include "trace.h"
+
+#define min(a,b)            (((a) < (b)) ? (a) : (b))
+
+/* 
+ * Macros for user secret regex
+ * These are BRE regex, please refer to: https://en.wikibooks.org/wiki/Regular_Expressions/POSIX_Basic_Regular_Expressions
+ */
+#define USER_SECRET_REGEX_WHITE_SPACE              "[[:space:]]*"
+#define USER_SECRET_REGEX_SECRET                   "\\([^[:space:]]*\\)"

```

Comment:
The macro names are not easy to understand. I guess you mean `REGEX_WHITESPACES`, `REGEX_TOKEN`, or something similar.

`user secret` is really confusing.

Reply:
Fixed by rename to password.

Reply:
password does not fit exactly.

Reply:
Fixed by rename to REGEX_WHITESPACES and REGEX_TOKEN